bigchaindb/consensus.py

```diff
@@ -19,10 +26,19 @@ def validate_block(bigchain, block):
         return block.validate(bigchain)
 
     @staticmethod
-    def verify_vote_signature(voters, signed_vote):
+    def verify_vote(voters, signed_vote):
         """Verify the signature of a vote.
 
         Refer to the documentation of
         :func:`bigchaindb.util.verify_signature`.
         """
-        return verify_vote_signature(voters, signed_vote)
+        if verify_vote_signature(voters, signed_vote):
+            try:
+                validate_vote_schema(signed_vote)
+                return True
+            except SchemaValidationError as exc:
+                logger.warning(exc)
+        else:
+            logger.warning("Vote failed signature verification: "
+                           "%s with voters: %s", signed_vote, voters)
+        return False
```

Comment:
It's kind of weird that this returns a bool status whereas the transaction validation throws... but that's a problem for another day.